src/tnlp.cc

```diff
@@ -61,13 +61,7 @@ namespace roboptim
       for (citer_t it = solver_.problem ().constraints ().begin ();
 	   it != solver_.problem ().constraints ().end (); ++it)
 	{
-	  shared_ptr<solver_t::commonConstraintFunction_t> g;
-	  if (it->which () == LINEAR)
-	    g = get<shared_ptr<linearFunction_t> > (*it);
-	  else
-	    g = get<shared_ptr<nonLinearFunction_t> > (*it);
-
-	  nnz_jac_g += g->jacobian (x).nonZeros ();
+	  nnz_jac_g += (*it)->castInto<GenericDifferentiableFunction<traits_t> >()->jacobian (x).nonZeros ();
```

Comment:
Wouldn't it be better to do all the casting in the initialization, and store vectors of pointers to the relevant types?